### PR TITLE
{AKS} Refactor a twelfth part of the create sub-command in acs module

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

* refactor api server access profile

Unit test result:
* "covered_lines": 949,
* "num_statements": 1008,
* "percent_covered": 94.14682539682539,
* "missing_lines": 59,
* "excluded_lines": 0

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 1]
```diff
@@ -2694,6 +2642,233 @@ def get_enable_azure_rbac(self, enable_validation: bool = False, **kwargs) -> bo
...
+                for x in (
+                    api_server_authorized_ip_ranges.split(",")
+                    if api_server_authorized_ip_ranges
+                    else []
+                )
+            ]
+
+        # validation
+        if api_server_authorized_ip_ranges:
+            if self._get_load_balancer_sku(enable_validation=False) == "basic":
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:2684]: use ignore case equal?

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 2]
```diff
@@ -2694,6 +2642,233 @@ def get_enable_azure_rbac(self, enable_validation: bool = False, **kwargs) -> bo
...
+            self.mc.api_server_access_profile.enable_private_cluster is not None
+        ):
+            enable_private_cluster = self.mc.api_server_access_profile.enable_private_cluster
+
+        # this parameter does not need dynamic completion
+
+        # validation
+        if enable_validation:
+            if enable_private_cluster:
+                if self._get_load_balancer_sku(enable_validation=False) == "basic":
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:2773]: ignore case

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 3]
```diff
@@ -3193,6 +3369,34 @@ def set_up_aad_profile(self, mc) -> ManagedCluster:
         mc.aad_profile = aad_profile
         return mc
 
+    def set_up_api_server_access_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        """Set up api server access profile and fqdn subdomain for the ManagedCluster object.
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:3373]: suggest to put the unit test coverage data for the PR.

**Review by andyliuliming — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py (hunk 3 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py:3373]: Sure, updated in the description.

Unit test result:
- "covered_lines": 949,
- "num_statements": 1008,
- "percent_covered": 94.14682539682539,
- "missing_lines": 59,
- "excluded_lines": 0

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py (hunk 1 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py]: Updated in commit [#62fd075](https://github.com/Azure/azure-cli/pull/19539/commits/62fd075d4d50739e6b1e47fa81de0ed79dcdf962).

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py (hunk 2 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py]: Updated in commit [#62fd075](https://github.com/Azure/azure-cli/pull/19539/commits/62fd075d4d50739e6b1e47fa81de0ed79dcdf962).

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**